# -*- coding: utf-8 -*-

#A. Thomer - modding from spring 2013 text mining code for use in paratext project

#python 2.7.2 or NLTK won't work

tester = """
<!DOCTYPE article PUBLIC "-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD v1.0 20120330//EN" "JATS-archivearticle1.dtd">
<article xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:mml="http://www.w3.org/1998/Math/MathML" article-type="product-review"><?properties open_access?><front><journal-meta><journal-id journal-id-type="nlm-ta">BMC Bioinformatics</journal-id><journal-title-group><journal-title>BMC Bioinformatics</journal-title></journal-title-group><issn pub-type="epub">1471-2105</issn><publisher><publisher-name>BioMed Central</publisher-name></publisher></journal-meta><article-meta><article-id pub-id-type="pmid">21851628</article-id><article-id pub-id-type="pmc">3179970</article-id><article-id pub-id-type="publisher-id">1471-2105-12-348</article-id><article-id pub-id-type="doi">10.1186/1471-2105-12-348</article-id><article-categories><subj-group subj-group-type="heading"><subject>Software</subject></subj-group></article-categories><title-group><article-title>DARS-RNP and QUASI-RNP: New statistical potentials for protein-RNA docking</article-title></title-group><contrib-group><contrib contrib-type="author" id="A1"><name><surname>Tuszynska</surname><given-names>Irina</given-names></name><xref ref-type="aff" rid="I1">1</xref><email>irena@genesilico.pl</email></contrib><contrib contrib-type="author" corresp="yes" id="A2"><name><surname>Bujnicki</surname><given-names>Janusz M</given-names></name><xref ref-type="aff" rid="I1">1</xref><xref ref-type="aff" rid="I2">2</xref><email>iamb@genesilico.pl</email></contrib></contrib-group><aff id="I1"><label>1</label>Laboratory of Bioinformatics and Protein Engineering, International Institute of Molecular and Cell Biology, ul. Ks. Trojdena 4, PL-02-109 Warsaw, Poland</aff><aff id="I2"><label>2</label>Institute of Molecular Biology and Biotechnology, Adam Mickiewicz University, ul. Umultowska 89, PL-61-614 Poznan, Poland</aff><pub-date pub-type="collection"><year>2011</year></pub-date><pub-date pub-type="epub"><day>18</day><month>8</month><year>2011</year></pub-date><volume>12</volume><fpage>348</fpage><lpage>348</lpage><history><date date-type="received"><day>31</day><month>3</month><year>2011</year></date><date date-type="accepted"><day>18</day><month>8</month><year>2011</year></date></history><permissions><copyright-statement>Copyright &#x000a9;2011 Tuszynska and Bujnicki; licensee BioMed Central Ltd.</copyright-statement><copyright-year>2011</copyright-year><copyright-holder>Tuszynska and Bujnicki; licensee BioMed Central Ltd.</copyright-holder><license license-type="open-access" xlink:href="http://creativecommons.org/licenses/by/2.0"><license-p>This is an Open Access article distributed under the terms of the Creative Commons Attribution License (<ext-link ext-link-type="uri" xlink:href="http://creativecommons.org/licenses/by/2.0">http://creativecommons.org/licenses/by/2.0</ext-link>), which permits unrestricted use, distribution, and reproduction in any medium, provided the original work is properly cited.</license-p></license></permissions><self-uri xlink:href="http://www.biomedcentral.com/1471-2105/12/348"/><abstract><sec><title>Background</title><p>Protein-RNA interactions play fundamental roles in many biological processes. Understanding the molecular mechanism of protein-RNA recognition and formation of protein-RNA complexes is a major challenge in structural biology. Unfortunately, the experimental determination of protein-RNA complexes is tedious and difficult, both by X-ray crystallography and NMR. For many interacting proteins and RNAs the individual structures are available, enabling computational prediction of complex structures by computational docking. However, methods for protein-RNA docking remain scarce, in particular in comparison to the numerous methods for protein-protein docking.</p></sec><sec><title>Results</title><p>We developed two medium-resolution, knowledge-based potentials for scoring protein-RNA models obtained by docking: the quasi-chemical potential (QUASI-RNP) and the Decoys As the Reference State potential (DARS-RNP). Both potentials use a coarse-grained representation for both RNA and protein molecules and are capable of dealing with RNA structures with posttranscriptionally modified residues. We compared the discriminative power of DARS-RNP and QUASI-RNP for selecting rigid-body docking poses with the potentials previously developed by the Varani and Fernandez groups.</p></sec><sec><title>Conclusions</title><p>In both bound and unbound docking tests, DARS-RNP showed the highest ability to identify native-like structures. Python implementations of DARS-RNP and QUASI-RNP are freely available for download at <ext-link ext-link-type="uri" xlink:href="http://iimcb.genesilico.pl/RNP/">http://iimcb.genesilico.pl/RNP/</ext-link></p></sec></abstract><kwd-group><kwd>RNA</kwd><kwd>protein</kwd><kwd>RNP</kwd><kwd>macromolecular docking</kwd><kwd>complex modeling</kwd><kwd>structural bioinformatics</kwd></kwd-group></article-meta></front><body><sec><title>Background</title><p>Protein-RNA interactions play fundamental roles in many biological processes, such as regulation of gene expression, RNA splicing, protein synthesis, replication of viral RNAs, and virus assembly (review: [<xref ref-type="bibr" rid="B1">1</xref>]). Defects of protein-RNA interactions are responsible for many human diseases ranging from neurological disorders to cancer [<xref ref-type="bibr" rid="B2">2</xref>]. The understanding of these processes improves as new structures of protein-RNA complexes are solved and the molecular details of interactions analyzed. Unfortunately, the experimental determination of protein-RNA complexes is a slow and difficult process [<xref ref-type="bibr" rid="B3">3</xref>,<xref ref-type="bibr" rid="B4">4</xref>]. The ability to predict structures of protein-RNA complexes computationally would greatly help us study protein-RNA interactions. However, while there is a growing number of methods for modeling protein and RNA structures (reviews: [<xref ref-type="bibr" rid="B5">5</xref>,<xref ref-type="bibr" rid="B6">6</xref>]), the number of methods for modeling protein-RNA complexes remains scarce.</p><p>Docking methods are widely used to predict structures of macromolecular complexes, starting from structures of the individual components [<xref ref-type="bibr" rid="B7">7</xref>]. All docking methods face two main challenges: to search the space of possible orientations and conformations (poses) of the components and to identify near-native structures among the alternative complex models (decoys) generated. An ideal docking method should be able to reliably reconstruct a native complex from its 'bound' components, and score it significantly better than any non-native decoys. In real life, the structure of the complex is unknown, and while the structures of binding partners are solved in isolation, the task of the 'unbound' docking experiment is not only to assemble them into a complex, but also to take into account possible conformational changes upon binding. Conformational changes are either modeled explicitly at the atomic level (which makes such analyses computationally very demanding), or a certain level of 'fuzziness' is introduced e.g. by allowing for some extent of steric overlaps between atoms or by 'coarse-graining' of the representation i.e. by neglecting some atoms or grouping them into 'united atoms' to be considered jointly (reviews: [<xref ref-type="bibr" rid="B7">7</xref>,<xref ref-type="bibr" rid="B8">8</xref>]).</p><p>One interesting and frequently neglected aspect of RNA structure and interactions is the presence of posttranscriptional modifications, which increase the basic set of four nucleotides (A, U, G, C) to more than 100 variants with altered base and/or ribose moieties [<xref ref-type="bibr" rid="B9">9</xref>]. Modified residues in RNA are involved in many processes, including RNA folding and RNA-RNA interactions (reviews: [<xref ref-type="bibr" rid="B10">10</xref>]), but also specific RNA-protein recognition and binding [<xref ref-type="bibr" rid="B11">11</xref>,<xref ref-type="bibr" rid="B12">12</xref>]. To our knowledge, among freely available macromolecular docking methods only three are suitable for handling post-transcriptional nucleotide modifications in RNA without the need of 'demodification'. HADDOCK accepts RNA as a part of the complex to be predicted, but the user is required to provide force field parameters for all modified nucleotides [<xref ref-type="bibr" rid="B13">13</xref>]. GRAMM [<xref ref-type="bibr" rid="B14">14</xref>] and HEX [<xref ref-type="bibr" rid="B15">15</xref>] can also perform protein-RNA docking for RNA structures with modified nucleotides, but the scoring functions of these programs have been developed to evaluate protein-protein complexes, and while they can generate the poses for protein-RNA complexes, they are unable to identify near-native variants from a set of decoys. A useful extension of the latter kind of methods would be the development of a potential for scoring protein-RNA complexes.</p><p>Recently, new statistical potentials for scoring protein-RNA complexes have been proposed: a distance-dependent all-atom potential developed by the Varani group [<xref ref-type="bibr" rid="B16">16</xref>], and a residue-level potential developed by the Fernandez group [<xref ref-type="bibr" rid="B17">17</xref>]. The Varani potential performs well in discriminating models of protein-RNA complexes that are very close to the native structure, i.e. with the root mean square deviation (RMSD) &#x0003c; 5 &#x000c5; [<xref ref-type="bibr" rid="B16">16</xref>]. However, during a real (unbound) docking experiment it is difficult to obtain many decoys with RMSD &#x0003c; 5 &#x000c5;. The Fernandez potential was designed to improve the discriminative power of the FTDock potential and is not available as a standalone program. The FTDock program [<xref ref-type="bibr" rid="B18">18</xref>] was developed for protein-protein and protein-DNA docking, but it accepts only conventional RNA molecules (without modified nucleotides).</p><p>In this article we introduce two new, medium-resolution, knowledge-based potentials for scoring protein-RNA models: the quasi-chemical potential (QUASI-RNP) and the Decoys As the Reference State [<xref ref-type="bibr" rid="B19">19</xref>] potential (DARS-RNP). These potentials are based on a reduced representation of protein and RNA, use the same mathematical base but differ in their reference state.</p><p>We compare the discriminative power of our new statistical potentials to the Varani and Fernandez potentials. For the bound docking test set our potentials discriminated native-like (with RMSD &#x0003c; 10 &#x000c5;) structures of protein-RNA complexes, the potential developed by the Varani group recognized structures very close to the native (RMSD &#x0003c; 5 &#x000c5;), whereas the Fernandez potential recognized near native structures only for some protein-RNA complexes. For the unbound docking test set, our potentials have the highest discriminative ability of alternative models. Our new knowledge-based potentials are a useful tool for scoring protein-RNA complexes generated by macromolecular docking methods, such as GRAMM or Hex.</p><sec><title>Implementation</title><sec><title>Docking</title><p>To search the space of possible orientations and conformations of the components, we employed the GRAMM method for medium- to low-resolution docking [<xref ref-type="bibr" rid="B20">20</xref>]. As opposed to high-resolution methods that typically operate in the continuous space, GRAMM discretizes the system (thereby lowers its resolution) by projecting the macromolecular structures on a grid and allows for imprecise fit by 'softening' the van der Waals interactions and permitting some degree of steric conflicts. One of the components of a binary complex, referred to as the "receptor", is fixed, while the other component, referred to as the "ligand", is rotated and translated around the receptor to obtain geometrically compatible poses.</p><p>The van der Waals radius was used as a projection of an atom. The value of the grid step was set to 1.7 &#x000c5; for complexes from the bound docking set and to the minimal value allowed by the program for complexes from the unbound docking set, a repulsion parameter to 10 (attraction is always -1) and attraction double range to 0. Ligand structures were rotated by 10&#x000b0; angle intervals. If significant steric clashes were observed in a large fraction of models obtained in preliminary GRAMM runs, as assessed by visual inspection, the repulsion parameter was increased to decrease the volume of steric conflict, until docking decoys reached physically reasonable geometry (Additional file <xref ref-type="supplementary-material" rid="S1">1</xref>, Table S1). We defined "native-like" poses as those with the ligand RMSD &#x0003c; 10 &#x000c5; from the native complex structure. According to our experience, this value is appropriate for consideration of medium- and low-resolution docking experiments. The distance between the experimentally determined and decoy complex structure was calculated as the RMSD of all heavy atoms of ligands, following optimal superposition of the receptor structures.</p><p>The resulting sets of decoys are supposed to approximate a broad distribution of structures that exhibit relatively good spatial complementarity between the receptor and the ligand, regardless of other interactions that contribute to the observed strength of binding. Such sets may be enriched into native-like decoys comparably to the completely random models (and in fact, in some cases they are), but "spatially complementary" models are much better approximation of the real-life sampling. The false models were not selected by the GRAMM scores, as we found that the GRAMM score alone is a poor predictor of the complex native-likeness (data not shown). We treated all 10000 decoys per complex as equal, both for training of our potential, and for evaluation with all four potentials analyzed in this work.</p></sec><sec><title>Statistical potentials used</title><p>The distance and orientation-dependent knowledge-based potentials described in this article (DARS-RNP and QUASI-RNP) were generated using reverse Boltzmann statistics, where the interaction energy <italic>&#x003b5; </italic>between the united atom type from the protein <italic>i </italic>and the united atom type from the RNA <italic>j </italic>is calculated according to Formula (1)</p><p><disp-formula id="bmcM1"><label>(1)</label><mml:math id="M1" name="1471-2105-12-348-i1" overflow="scroll"><mml:mrow><mml:mi>&#x003f5;</mml:mi><mml:mfenced open="(" close=")"><mml:mrow><mml:mstyle class="text"><mml:mtext class="textsf" mathvariant="sans-serif">i,j,d</mml:mtext></mml:mstyle></mml:mrow></mml:mfenced><mml:mo class="MathClass-rel">=</mml:mo><mml:mo class="MathClass-bin">-</mml:mo><mml:mi>R</mml:mi><mml:mi>T</mml:mi><mml:mstyle mathvariant="italic"><mml:mi>l</mml:mi><mml:mi>n</mml:mi></mml:mstyle><mml:mfrac><mml:mrow><mml:mi>N</mml:mi><mml:mi>o</mml:mi><mml:mi>b</mml:mi><mml:mi>s</mml:mi><mml:mfenced open="(" close=")"><mml:mrow><mml:mi>i</mml:mi><mml:mo class="MathClass-punc">,</mml:mo><mml:mi>j</mml:mi><mml:mo class="MathClass-punc">,</mml:mo><mml:mi>d</mml:mi></mml:mrow></mml:mfenced></mml:mrow><mml:mrow><mml:mi>N</mml:mi><mml:mstyle mathvariant="italic"><mml:mi>e</mml:mi><mml:mi>x</mml:mi><mml:mi>p</mml:mi></mml:mstyle><mml:mfenced open="(" close=")"><mml:mrow><mml:mi>i</mml:mi><mml:mo class="MathClass-punc">,</mml:mo><mml:mi>j</mml:mi><mml:mo class="MathClass-punc">,</mml:mo><mml:mi>d</mml:mi></mml:mrow></mml:mfenced></mml:mrow></mml:mfrac></mml:mrow></mml:math></disp-formula></p><p>where <italic>R </italic>is the gas constant, <italic>T </italic>is the temperature, <italic>N<sub>obs </sub>(i, j, d) </italic>is the number of contacts between atom types <italic>i </italic>and <italic>j </italic>observed in the training set in a distance/angular bin <italic>d</italic>, while <italic>N<sub>exp </sub>(i, j, d) </italic>is the expected number of contacts at the same distance/angle in the reference state. There are two types of bins, connected with different terms of the potential: a distance bin (1 &#x000c5;) used in the distance-dependent term, and an angular bin (20&#x000b0;) used in the angular-dependent term. The energy is calculated for each pair of protein-RNA united atoms that are within the distance of 9 &#x000c5; from each other. This threshold parameter was chosen based on the analysis carried out by the Shakhnovich group [<xref ref-type="bibr" rid="B21">21</xref>], who identified 7 &#x000c5; as the optimal threshold for evaluating protein-nucleic acid interactions with all-atom multiple bin potentials. For our reduced representation potentials we have tested five distance thresholds (7 &#x000c5;, 8 &#x000c5;, 9 &#x000c5;, 10 &#x000c5; and 11 &#x000c5;) by calculating correlation coefficients and a position of the native structure in the decoys ranking. The best results was for 9 &#x000c5; for both DARS and QUASI potentials (slightly better than for 7 and 8), and it was not statistically distinguishable from 10 &#x000c5; and 11 &#x000c5; (data not shown).</p><p>Our two statistical potentials calculate the reference state in different ways. The reference state for the QUASI-RNP potential is calculated using mole fractions of residues:</p><p><disp-formula id="bmcM2"><label>(2)</label><mml:math id="M2" name="1471-2105-12-348-i2" overflow="scroll"><mml:mrow><mml:msub><mml:mrow><mml:mi>N</mml:mi></mml:mrow><mml:mrow><mml:mo class="qopname">exp</mml:mo></mml:mrow></mml:msub><mml:mfenced open="(" close=")"><mml:mrow><mml:mi>i</mml:mi><mml:mo class="MathClass-punc">,</mml:mo><mml:mi>j</mml:mi><mml:mo class="MathClass-punc">,</mml:mo><mml:mi>d</mml:mi></mml:mrow></mml:mfenced><mml:mo class="MathClass-rel">=</mml:mo><mml:msub><mml:mrow><mml:mi>X</mml:mi></mml:mrow><mml:mrow><mml:mi>i</mml:mi></mml:mrow></mml:msub><mml:mo class="MathClass-bin">*</mml:mo><mml:msub><mml:mrow><mml:mi>X</mml:mi></mml:mrow><mml:mrow><mml:mi>j</mml:mi></mml:mrow></mml:msub><mml:mo class="MathClass-bin">*</mml:mo><mml:msub><mml:mrow><mml:mi>N</mml:mi></mml:mrow><mml:mrow><mml:mi>o</mml:mi><mml:mi>b</mml:mi><mml:mi>s</mml:mi></mml:mrow></mml:msub><mml:mfenced open="(" close=")"><mml:mrow><mml:mi>d</mml:mi></mml:mrow></mml:mfenced><mml:mo class="MathClass-punc">,</mml:mo></mml:mrow></mml:math></disp-formula></p><p>where <italic>X<sub>i </sub></italic>and <italic>X<sub>j </sub></italic>are the mole fractions of atom type <italic>i </italic>and <italic>j </italic>respectively, while <italic>N<sub>obs </sub>(d) </italic>is the total number of contacts in bin <italic>d</italic>. On the other hand, in the DARS approach [<xref ref-type="bibr" rid="B19">19</xref>]<italic>N<sub>exp </sub>(i, j, d) </italic>is a normalized number of contacts between atom types i and j in a set of decoy structures that are considered as random models. 1000 decoys were generated for each native structure of protein-RNA complex from the training set by using the GRAMM docking program [<xref ref-type="bibr" rid="B20">20</xref>], with the following parameters: values for repulsion and attraction were 15 and 0, respectively, while a grid step used the minimal value allowed by the program, depending on the size of the components to be docked. In a few cases where default values led to generation of decoys with significant steric clashes, the repulsion value was increased stepwise, until physically realistic decoys were obtained. It must be emphasized that these decoys maximize the geometric fit between the protein and RNA molecule, but do not take any interaction energies into account.</p><p>Both statistical potentials comprise a distance-dependent energy term (E<sub>d</sub>), an angular-dependent energy term (E<sub>a</sub>), a site-dependent energy term (E<sub>s</sub>), and a penalty for steric clashes (E<sub>p</sub>) (Figure <xref ref-type="fig" rid="F1">1</xref>):</p><fig id="F1" position="float"><label>Figure 1</label><caption><p><bold>Schematic representation of four terms used in DARS-RNP and QUASI-RNP statistical potentials: A, a distance-dependent term; B, an angular-dependent term; C, an edge-dependent term as a site term; D, a penalty for steric clashes</bold>. United atoms of nucleotide and amino acid residues are colored in dark and light gray, respectively.</p></caption><graphic xlink:href="1471-2105-12-348-1"/></fig><p><disp-formula id="bmcM3"><label>(3)</label><mml:math id="M3" name="1471-2105-12-348-i3" overflow="scroll"><mml:mrow><mml:mstyle class="text"><mml:mtext class="textsf" mathvariant="sans-serif">E</mml:mtext></mml:mstyle><mml:mo class="MathClass-rel">=</mml:mo><mml:mstyle class="text"><mml:mtext class="textsf" mathvariant="sans-serif">Ed</mml:mtext></mml:mstyle><mml:mo class="MathClass-bin">+</mml:mo><mml:mstyle class="text"><mml:mtext class="textsf" mathvariant="sans-serif">Ea</mml:mtext></mml:mstyle><mml:mo class="MathClass-bin">+</mml:mo><mml:mstyle class="text"><mml:mtext class="textsf" mathvariant="sans-serif">Es</mml:mtext></mml:mstyle><mml:mo class="MathClass-bin">+</mml:mo><mml:mstyle class="text"><mml:mtext class="textsf" mathvariant="sans-serif">Ep</mml:mtext></mml:mstyle></mml:mrow></mml:math></disp-formula></p><p>The site-dependent term assesses the probability of interaction of amino acid residues with edges of nucleotide residues: Watson-Crick, Sugar and Hoogsteen, as defined by Leontis at.al [<xref ref-type="bibr" rid="B22">22</xref>] (Figure <xref ref-type="fig" rid="F1">1</xref>).</p><p>All four terms of the energy function exhibit comparable values (see Figure <xref ref-type="fig" rid="F2">2</xref> and <xref ref-type="fig" rid="F3">3</xref> for graphical presentation of examples and Additional File <xref ref-type="supplementary-material" rid="S2">2</xref> and Additional File <xref ref-type="supplementary-material" rid="S3">3</xref> for the list of all values). Hence, we assigned equal weights to all four terms and have not optimized them further in any way. Additional file <xref ref-type="supplementary-material" rid="S1">1</xref>, Figures S5, S6 and S7 show selected graphs of N<sub>obs </sub>common for DARS-RNP and QUASI-RNP, N<sub>exp </sub>for DARS-RNP, and N<sub>exp </sub>for QUASI-RNP, respectively, as a function of distance, angle and nucleotide edges, while Additional Files <xref ref-type="supplementary-material" rid="S4">4</xref>, <xref ref-type="supplementary-material" rid="S5">5</xref> and <xref ref-type="supplementary-material" rid="S6">6</xref> include the list of values for all cases.</p><fig id="F2" position="float"><label>Figure 2</label><caption><p><bold>Examples of value distributions for three terms of the DARS-RNP potential that describe interactions between six arbitrarily selected pars of united atoms</bold>.</p></caption><graphic xlink:href="1471-2105-12-348-2"/></fig><fig id="F3" position="float"><label>Figure 3</label><caption><p><bold>Examples of value distributions for three terms of the QUASI-RNP potential that describe interactions between six arbitrarily selected pars of united atoms</bold>.</p></caption><graphic xlink:href="1471-2105-12-348-3"/></fig></sec><sec><title>Representation of molecules</title><p>To calculate the parameters of QUASI-RNP and DARS-RNP potentials, the all-atom representation of all macromolecular structures was transformed into a reduced representation. The atoms of each amino acid or nucleotide residue were replaced by a number of united atoms depending on the residue type. For amino acid residues, we used the representation used in the REFINER program [<xref ref-type="bibr" rid="B23">23</xref>,<xref ref-type="bibr" rid="B24">24</xref>], which involves one to three united atoms per residue, depending on the size of the residue. For nucleotide residues we employed the representation used in the RedRNA method, currently under development in our laboratory (Michal Boniecki and J.M.B. unpublished data). The RNA backbone was represented by two united atoms, one for the phosphate group (P) and one for the ribose (RIB), while the pyrimidine and purine rings were represented by one and two atoms, respectively (Figure <xref ref-type="fig" rid="F1">1</xref>). All united atoms and alpha carbons from the reduced representation in every residue were considered for the potentials as separate atom types, e.g. the alpha carbon of alanine and alpha carbon of lysine had different types, because they represented different type of residues.</p></sec><sec><title>Training set for developing the statistical potentials</title><p>In order to compare our QUASI-RNP and DARS-RNP potentials to the previously published Varani potential we used the same training set as the Varani group [<xref ref-type="bibr" rid="B16">16</xref>]. There are 72 protein-RNA complexes in the training set taken from crystal structures of protein-RNA complexes downloaded from the Protein Data Bank (PDB codes: <ext-link ext-link-type="pdb" xlink:href="1a1v">1a1v</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1a34">1a34</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1b2m">1b2m</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1c0a">1c0a</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1d9d">1d9d</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1dfu">1dfu</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1di2">1di2</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1bu1">1bu1</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1ec6">1ec6</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1f7u">1f7u</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1feu">1feu</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1ffy">1ffy</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1fxl">1fxl</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1gtf">1gtf</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1gtr">1gtr</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1hq1">1hq1</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1j1u">1j1u</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1jbs">1jbs</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1jid">1jid</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1jj2">1jj2</ext-link> - 50S ribosome structure, <ext-link ext-link-type="pdb" xlink:href="1k8w">1k8w</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1knz">1knz</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1lng">1lng</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1m8w">1m8w</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1mji">1mji</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1msw">1msw</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1n35">1n35</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1n78">1n78</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1ooa">1ooa</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1qln">1qln</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1r3e">1r3e</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1r9f">1r9f</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1rc7">1rc7</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1sds">1sds</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1tfw">1tfw</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1u0b">1u0b</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1urn">1urn</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1uvj">1uvj</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1wsu">1wsu</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1yvp">1yvp</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1zbi">1zbi</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1a8v">1a8v</ext-link>, <ext-link ext-link-type="pdb" xlink:href="2bgg">2bgg</ext-link>, <ext-link ext-link-type="pdb" xlink:href="2bh2">2bh2</ext-link>) [<xref ref-type="bibr" rid="B25">25</xref>]. One of these is the 50S ribosomal subunit, which comprises of 28 individual peptide chains in complex with RNA. Due to the limited number of protein-RNA complexes in this set, we performed a "leave one out" cross-validation, in which the potential was recalculated for testing of each structure, based on a training set with the tested structure excluded.</p><p>Our software uses an in-house modified variant of PDBParser from Biopython (BrutePDBParser developed by Michal Pietal in IIMCB) to parse PDB files and ignore information about e.g. atom occupancy.</p></sec><sec><title>Testing statistical potentials for protein-RNA docking</title><p>To test the discriminatory power of the QUASI-RNP and DARS-RNP knowledge-based potentials we compared them with two existing statistical potentials developed by the Varani [<xref ref-type="bibr" rid="B16">16</xref>] and Fernandez [<xref ref-type="bibr" rid="B17">17</xref>] groups. Software to calculate the Varani potential has been kindly provided by the Author (Gabriele Varani, personal communication). Since the Fernandez potential is not available as a standalone software, the Authors have kindly provided raw statistical data for each amino acid-nucleotide pair calculated based on their training set (Juan Fernandez-Recio, personal communication), which we used to calculate our local implementation of their scoring function, independent of the FTDock program.</p><p>Two types of test sets were used, based on bound structures (components isolated from complex structures), or on unbound structures (counterparts of complex components, in which one structure or both were solved in isolation from each other).</p><sec><title>Bound docking test sets</title><p>We used one set of molecules with unmodified (bound) conformations of entire RNA molecules and protein backbones, but with optimized protein side chains. This set was developed by the Varani group to perform decoy-discrimination tests of their all-atom potentials [<xref ref-type="bibr" rid="B16">16</xref>,<xref ref-type="bibr" rid="B26">26</xref>]. Their decoys were obtained by modifying five native protein-RNA complexes (PDB codes: <ext-link ext-link-type="pdb" xlink:href="1cvj">1cvj</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1ec6">1ec6</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1fxl">1fxl</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1jid">1jid</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1urn">1urn</ext-link>) using the docking module of ROSETTA, through the use of the protein side chain repacking algorithm [<xref ref-type="bibr" rid="B27">27</xref>,<xref ref-type="bibr" rid="B28">28</xref>]. For each protein complex they have generated 2000 structures with the RMSD from the native complex structure ranging from 0.2 &#x000c5; to about 30 &#x000c5;. The RMSD of two complexes was calculated based on ligand heavy atoms, after superposition of receptors.</p><p>The second set contains all molecules in unmodified (bound) conformations, and was generated by ourselves using the high-resolution mode of the GRAMM docking program. For each of the five protein-RNA complexes from the Varani test set 10,000 alternative docking decoys were generated according to the procedure described above in the Docking section.</p></sec><sec><title>Unbound docking test set</title><p>The unbound docking test set was based on twelve native protein-RNA complexes (PDB codes: <ext-link ext-link-type="pdb" xlink:href="2rkj">2rkj</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1wsu">1wsu</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1ooa">1ooa</ext-link>, <ext-link ext-link-type="pdb" xlink:href="2r8s">2r8s</ext-link>, <ext-link ext-link-type="pdb" xlink:href="2pjp">2pjp</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1lng">1lng</ext-link>, <ext-link ext-link-type="pdb" xlink:href="2pxv">2pxv</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1e7k">1e7k</ext-link>, <ext-link ext-link-type="pdb" xlink:href="1wpu">1wpu</ext-link>, <ext-link ext-link-type="pdb" xlink:href="3bso">3bso</ext-link>, <ext-link ext-link-type="pdb" xlink:href="2qux">2qux</ext-link>, <ext-link ext-link-type="pdb" xlink:href="2jea">2jea</ext-link>), previously used by the Fernandez group to test their potential [<xref ref-type="bibr" rid="B17">17</xref>]. For each component of these complexes, at least one independently solved 3D structure per complex is available. The GRAMM program was used to generate 10,000 docking decoys for each complex. We used the same parameters as in the bound docking procedure. With these settings, the GRAMM program generated at least five native-like structures (RMSD &#x0003c; 10 &#x000c5;) for eight out of twelve protein-RNA complexes. Only these eight decoy sets with native-like structures were considered.</p></sec></sec><sec><title>Clustering the best scored decoys</title><p>Critical assessment of protein structure predictions (in particular the CASP experiment) has demonstrated that the scoring functions alone may not be the best discriminators of native-like structures, and better results may be achieved by clustering well-scored suboptimal structures [<xref ref-type="bibr" rid="B29">29</xref>,<xref ref-type="bibr" rid="B30">30</xref>]. We have applied this approach, in particular using the clustering algorithm proposed by Baker and coworkers [<xref ref-type="bibr" rid="B31">31</xref>], which has worked very well in protein structure prediction. First, an RMSD is calculated for all pairs of structures and stored in a distance matrix. Then, the row of the distance matrix with the largest number of RMSD values smaller than a cutoff (default 5 &#x000c5;) is selected. Structures in that row with RMSD below the cutoff value are assigned to one cluster and excluded from the matrix. The process is iterated until all structures with RMSD smaller than the cutoff are assigned to clusters. Three biggest clusters are then considered as candidates for groups of potentially native-like structures, from which then the lowest-energy decoys are identified. As a rule of thumb, the larger the cluster and the better score of its decoys - the higher chance that it contains a native-like structure.</p></sec></sec></sec><sec><title>Results</title><p>In order to compare the ability to identify near-native protein-RNA complex structures by our two potentials (QUASI-RNP and DARS-RNP), the potential developed by Varani et al. [<xref ref-type="bibr" rid="B16">16</xref>], and the potential developed by Fernandez et al. [<xref ref-type="bibr" rid="B17">17</xref>], we used test sets, obtained by bound and unbound docking. To give all methods equal chances in finding structures close to the native one, we included the clustering procedure implemented in our potentials as an additional step in the scoring performed by both the Varani and Fernandez potentials. We have calculated correlation coefficients for decoys with RMSD in the range of 0-5 &#x000c5;, 0-10 &#x000c5; and 0-20 &#x000c5;. The RMSD range &#x0003c; 5 &#x000c5; was chosen because the Varani potential has a recognition area of near-native structures around 5 &#x000c5;. The RMSD threshold 10 &#x000c5; is a "golden standard" of near to native decoys definition used in the CAPRI (Critical Assessment of PRediction of Interactions) experiment. The generous 20 &#x000c5; threshold was used because we observed that for large structures there exist decoys with a biologically reasonable match of the binding sites and a number of native-like protein-RNA contacts, despite the global RMSD of the ligand in the range of 10-20 &#x000c5;. Such deviations are typically due to a rotation of the ligand that retains contacts in the binding site, but moves around atoms that are far away from the binding site, as shown e.g. in Additional file <xref ref-type="supplementary-material" rid="S1">1</xref>, Figures S1 and S2.</p><sec><title>Decoy discrimination for the bound docking test set</title><p>We tested the DARS-RNP, QUASI-RNP, Varani, and Fernandez potentials for scoring of RNA-protein complexes on two bound test sets, and examined which potential gives the best correlation coefficient between scores and RMSD of the corresponding decoys. For the Varani test set, the DARS-RNP and QUASI-RNP potentials recognized most structures with ligand RMSD &#x0003c; 10 &#x000c5;. We found strong correlation between DARS-RNP scores and RMSD, as well between QUASI-RNP scores and RMSD for models up to 10 &#x000c5; from the native structure (Figure <xref ref-type="fig" rid="F4">4</xref>, Table <xref ref-type="table" rid="T1">1</xref>). The Varani potential discriminates models with ligand RMSD &#x0003c; 5 &#x000c5;, but generally fails to distinguish structures with 5-10 &#x000c5; RMSD from the native and random structures with larger RMSD values (Figure <xref ref-type="fig" rid="F4">4</xref> and Additional file <xref ref-type="supplementary-material" rid="S1">1</xref>, Figure S3). The Fernandez potential fails to recognize individual near-native decoys from the Varani test set; and its scores exhibit positive correlation coefficients with RMSDs only for decoys from three complexes in five (Table <xref ref-type="table" rid="T1">1</xref>). The clustering procedure improves the results obtained by the Fernendez potential, as the biggest clusters contain structures with RMSD &#x0003c; 5 &#x000c5; for all complexes considered. Nonetheless, the native structures and structures with the smallest RMSD are scored too poorly by the Fernandez potential to be included in the top clusters for two of five complexes - 1URN and 1EC6 (Figure <xref ref-type="fig" rid="F4">4</xref>).</p><fig id="F4" position="float"><label>Figure 4</label><caption><p><bold>Score-RMSD dependence for the Varani bound docking set</bold>. The three best clusters (1<sup>st</sup>, 2<sup>nd</sup>, 3<sup>rd</sup>) are colored in orange, violet, and blue, and the corresponding points are marked as squares, triangles, and rounds, respectively.</p></caption><graphic xlink:href="1471-2105-12-348-4"/></fig><table-wrap id="T1" position="float"><label>Table 1</label><caption><p>Results of scoring the decoys in the Varani bound docking set</p></caption><table frame="hsides" rules="groups"><thead><tr><th align="center">Complex PDB code</th><th align="center">RMSD threshold (decoy vs native)</th><th align="center">Number of decoys below threshold<break/>(per 2,000)</th><th align="center" colspan="4">Correlation coefficient</th></tr><tr><th/><th/><th/><th colspan="4"><hr/></th></tr><tr><th/><th/><th/><th align="center">DARSI-RNP potential<break/>(std. error)</th><th align="center">QUASI-RNP potential<break/>(std. error)</th><th align="center">Varani potential<break/>(std. error)</th><th align="center">Fernandez potential<break/>(std. error)</th></tr></thead><tbody><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="1urn">1urn</ext-link></td><td align="center">5</td><td align="center">726</td><td align="center">0.77 (0.024)</td><td align="center">0.7 (0.027)</td><td align="center">0.37 (0.035)</td><td align="center">-0.56 (0.031)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">10</td><td align="center">1457</td><td align="center">0.83 (0.015)</td><td align="center">0.79 (0.016)</td><td align="center">0.27 (0.025)</td><td align="center">-0.24 (0.025)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">20</td><td align="center">1967</td><td align="center">0.81 (0.013)</td><td align="center">0.79 (0.014)</td><td align="center">0.21 (0.022)</td><td align="center">-0.1 (0.022)</td></tr><tr><td colspan="7"><hr/></td></tr><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="1ec6">1ec6</ext-link></td><td align="center">5</td><td align="center">907</td><td align="center">0.81 (0.019)</td><td align="center">0.79 (0.020)</td><td align="center">0.57 (0.027)</td><td align="center">-0.08 (0.033)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">10</td><td align="center">1446</td><td align="center">0.9 (0.011)</td><td align="center">0.89 (0.012)</td><td align="center">0.38 (0.024)</td><td align="center">-0.05 (0.026)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">20</td><td align="center">1979</td><td align="center">0.87 (0.011)</td><td align="center">0.86 (0.011)</td><td align="center">0.31 (0.021)</td><td align="center">-0.02 (0.022)</td></tr><tr><td colspan="7"><hr/></td></tr><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="1fxl">1fxl</ext-link></td><td align="center">5</td><td align="center">881</td><td align="center">0.94 (0.012)</td><td align="center">0.95 (0.011)</td><td align="center">0.87 (0.017)</td><td align="center">0.61 (0.027)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">10</td><td align="center">1366</td><td align="center">0.96 (0.008)</td><td align="center">0.96 (0.008)</td><td align="center">0.82 (0.015)</td><td align="center">0.74 (0.018)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">20</td><td align="center">1758</td><td align="center">0.93 (0.009)</td><td align="center">0.94 (0.008)</td><td align="center">0.7 (0.017)</td><td align="center">0.83 (0.013)</td></tr><tr><td colspan="7"><hr/></td></tr><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="1cvj">1cvj</ext-link></td><td align="center">5</td><td align="center">936</td><td align="center">0.96 (0.009)</td><td align="center">0.96 (0.009)</td><td align="center">0.5 (0.028)</td><td align="center">0.85 (0.017)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">10</td><td align="center">1217</td><td align="center">0.97 (0.007)</td><td align="center">0.97 (0.007)</td><td align="center">0.61 (0.023)</td><td align="center">0.92 (0.011)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">20</td><td align="center">1947</td><td align="center">0.93 (0.008)</td><td align="center">0.93 (0.008)</td><td align="center">0.46 (0.020)</td><td align="center">0.9 (0.010)</td></tr><tr><td colspan="7"><hr/></td></tr><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="1jid">1jid</ext-link></td><td align="center">5</td><td align="center">828</td><td align="center">0.58 (0.028)</td><td align="center">0.58 (0.028)</td><td align="center">0.35 (0.033)</td><td align="center">0.33 (0.033)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">10</td><td align="center">1485</td><td align="center">0.72 (0.018)</td><td align="center">0.71 (0.018)</td><td align="center">0.3 (0.025)</td><td align="center">0.39 (0.024)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">20</td><td align="center">1978</td><td align="center">0.7 (0.016)</td><td align="center">0.69 (0.016)</td><td align="center">0.27 (0.022)</td><td align="center">0.29 (0.022)</td></tr><tr><td colspan="7"><hr/></td></tr><tr><td align="center">Mean</td><td align="center">5</td><td align="center">855.6</td><td align="center">0.81</td><td align="center">0.8</td><td align="center">0.53</td><td align="center">0.23</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">10</td><td align="center">1394.2</td><td align="center">0.88</td><td align="center">0.87</td><td align="center">0.47</td><td align="center">0.35</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">20</td><td align="center">1925.8</td><td align="center">0.85</td><td align="center">0.84</td><td align="center">0.39</td><td align="center">0.38</td></tr></tbody></table><table-wrap-foot><p>Correlation coefficients were calculated for scatter plots from Figure 4.</p></table-wrap-foot></table-wrap><p>In the bound docking test set generated by the GRAMM program, there is a smaller number of near-native structures than in the Varani set, and some of them exhibit steric clashes. For this test set, our DARS-RNP and QUASI-RNP potentials exhibit lower values of the correlation coefficient than for the Varani test set (Additional file <xref ref-type="supplementary-material" rid="S1">1</xref>, Figure S3). The Varani and Fernanez potentials discriminate GRAMM decoys better than Varani decoys for three complexes, and worse for two complexes (1urn and 1jid) (Figure <xref ref-type="fig" rid="F5">5</xref>) (Table <xref ref-type="table" rid="T1">1</xref> and Table <xref ref-type="table" rid="T2">2</xref>).</p><fig id="F5" position="float"><label>Figure 5</label><caption><p><bold>Score-RMSD dependence for the GRAMM-generated bound docking set</bold>. The three best clusters (1<sup>st</sup>, 2<sup>nd</sup>, 3<sup>rd</sup>) are colored in orange, violet, and blue, and the corresponding points are marked as squares, triangles, and rounds, respectively.</p></caption><graphic xlink:href="1471-2105-12-348-5"/></fig><table-wrap id="T2" position="float"><label>Table 2</label><caption><p>Results of scoring the decoys in the GRAMM-generated bound docking set</p></caption><table frame="hsides" rules="groups"><thead><tr><th align="center">Complex PDB code</th><th align="center">RMSD threshold (decoy vs native)</th><th align="center">Number of decoys below threshold<break/>(per 10,000)</th><th align="center" colspan="4">Correlation coefficient</th></tr><tr><th/><th colspan="6"><hr/></th></tr><tr><th/><th/><th/><th align="center">DARS-RNP potential<break/>(std. error)</th><th align="center">QUASI-RNP potential<break/>(std. error)</th><th align="center">Varani potential<break/>(std. error)</th><th align="center">Fernandez potential<break/>(std. error)</th></tr></thead><tbody><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="1urn">1urn</ext-link></td><td align="center">5</td><td align="center">21</td><td align="center">0.38 (0.212)</td><td align="center">0.25 (0.222)</td><td align="center">0.45 (0.205)</td><td align="center">-0.15 (0.227)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">10</td><td align="center">72</td><td align="center">0.51 (0.103)</td><td align="center">0.41 (0.109)</td><td align="center">0.46 (0.106)</td><td align="center">-0.01 (0.120)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">20</td><td align="center">352</td><td align="center">0.6 (0.043)</td><td align="center">0.45 (0.048)</td><td align="center">0.29 (0.051)</td><td align="center">-0.42 (0.049)</td></tr><tr><td colspan="7"><hr/></td></tr><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="1ec6">1ec6</ext-link></td><td align="center">5</td><td align="center">87</td><td align="center">0.3 (0.103)</td><td align="center">0.28 (0.104)</td><td align="center">0.4 (0.099)</td><td align="center">-0.12 (0.108)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">10</td><td align="center">155</td><td align="center">0.7 (0.058)</td><td align="center">0.68 (0.059)</td><td align="center">0.53 (0.069)</td><td align="center">0.03 (0.081)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">20</td><td align="center">1079</td><td align="center">0.65 (0.023)</td><td align="center">0.56 (0.025)</td><td align="center">0.31 (0.029)</td><td align="center">0.18 (0.030)</td></tr><tr><td colspan="7"><hr/></td></tr><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="1fxl">1fxl</ext-link></td><td align="center">5</td><td align="center">81</td><td align="center">0.62 (0.088)</td><td align="center">0.64 (0.086)</td><td align="center">0.62 (0.088)</td><td align="center">0.46 (0.100)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">10</td><td align="center">143</td><td align="center">0.81 (0.049)</td><td align="center">0.82 (0.048)</td><td align="center">0.61 (0.067)</td><td align="center">0.62 (0.066)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">20</td><td align="center">1732</td><td align="center">0.61 (0.019)</td><td align="center">0.52 (0.021)</td><td align="center">0.3 (0.023)</td><td align="center">0.29 (0.023)</td></tr><tr><td colspan="7"><hr/></td></tr><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="1cvj">1cvj</ext-link></td><td align="center">5</td><td align="center">38</td><td align="center">0.62 (0.131)</td><td align="center">0.49 (0.145)</td><td align="center">0.55 (0.139)</td><td align="center">0.34 (0.157)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">10</td><td align="center">76</td><td align="center">0.95 (0.036)</td><td align="center">0.91 (0.048)</td><td align="center">0.48 (0.102)</td><td align="center">0.72 (0.081)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">20</td><td align="center">1638</td><td align="center">0.49 (0.022)</td><td align="center">0.43 (0.022)</td><td align="center">0.18 (0.024)</td><td align="center">0.5 (0.021)</td></tr><tr><td colspan="7"><hr/></td></tr><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="1jid">1jid</ext-link></td><td align="center">5</td><td align="center">16</td><td align="center">0.13 (0.265)</td><td align="center">0.18 (0.263)</td><td align="center">0.73 (0.183)</td><td align="center">0.54 (0.225)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">10</td><td align="center">29</td><td align="center">0.13 (0.191)</td><td align="center">0.09 (0.192)</td><td align="center">0.59 (0.155)</td><td align="center">0.28 (0.185)</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">20</td><td align="center">175</td><td align="center">-0.06 (0.076)</td><td align="center">0.04 (0.076)</td><td align="center">0.58 (0.062)</td><td align="center">0.2 (0.074)</td></tr><tr><td colspan="7"><hr/></td></tr><tr><td align="center">Mean</td><td align="center">5</td><td align="center">49</td><td align="center">0.41</td><td align="center">0.37</td><td align="center">0.55</td><td align="center">0.22</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">10</td><td align="center">95</td><td align="center">0.62</td><td align="center">0.58</td><td align="center">0.54</td><td align="center">0.33</td></tr><tr><td/><td colspan="6"><hr/></td></tr><tr><td/><td align="center">20</td><td align="center">995</td><td align="center">0.46</td><td align="center">0.4</td><td align="center">0.33</td><td align="center">0.15</td></tr></tbody></table><table-wrap-foot><p>Correlation coefficients were calculated for scatter plots from Figure 5.</p></table-wrap-foot></table-wrap><p>Summarizing, the DARS-RNP and QUASI-RNP scores exhibit the highest correlation coefficients for all cutoffs in the Varani test set, and for 10 &#x000c5; and 20 &#x000c5; thresholds in the GRAMM test set (Additional file <xref ref-type="supplementary-material" rid="S1">1</xref>, Figure S3). Thus, DARS-RNP and QUASI-RNP potentials can be declared as "winners" of the bound docking test, except for the structures that are very close to the native complexes, where they are outperformed by the Varani potential.</p></sec><sec><title>Decoy discrimination for the unbound docking set</title><p>In an analogous way, we examined the discriminatory power of the DARS-RNP, QUASI-RNP, Varani, and Fernandez potentials for decoys of the unbound docking test set. The assessment of unbound docking results reveals, expectedly, that all potentials exhibit worse results than for the bound docking set (Figure <xref ref-type="fig" rid="F6">6</xref>). Here, the best results are obtained by our DARS-RNP potential, followed closely by the QUASI-RNP potential. These potentials recognized native-like structures for four out of eight complexes from the unbound test set (Figure <xref ref-type="fig" rid="F6">6</xref> and <xref ref-type="fig" rid="F7">7</xref>), while the Varani and Fernandez potentials recognized native-like structures only for one complex in this set with the default options of clustering, and two and three complexes respectively after increasing both number of clustering structures and RMSD threshold to 200 and 10 &#x000c5; (Figure <xref ref-type="fig" rid="F6">6</xref>). The correlation coefficients between the model score and RMSD are relatively low for the Varani potential (0.06, 0.06, and 0.01 for RMSD thresholds of 5, 10, and 20 &#x000c5;, respectively) and for the Fernandez potential (-0.13, -0.04 and 0.13), while the DARS-RNP/QUASI-RNP potentials exhibit correlation coefficients of 0.44/0.48, 0.25/0.23, and 0.37/0.33 for decoys with RMSD from the native structure lesser than 5, 10, and 20 &#x000c5;, respectively (Table <xref ref-type="table" rid="T3">3</xref>) and (Additional file <xref ref-type="supplementary-material" rid="S1">1</xref>, Figure S4).</p><fig id="F6" position="float"><label>Figure 6</label><caption><p><bold>Score-RMSD dependence for the GRAMM-generated unbound docking set as results for clustering of 100 best-scored docking decoys with the RMSD threshold of 5 &#x000c5;</bold>. * - no clusters found for 100 best scored decoys and 5 &#x000c5; threshold, results reported for 200 best scored decoys and the RMSD threshold of 10 &#x000c5;. The three best clusters (1<sup>st</sup>, 2<sup>nd</sup>, 3<sup>rd</sup>) are colored in red, blue, and orange, and the corresponding points are marked as squares, triangles, and rounds, respectively.</p></caption><graphic xlink:href="1471-2105-12-348-6"/></fig><fig id="F7" position="float"><label>Figure 7</label><caption><p><bold>Examples demonstrating the utility of clustering the best-scored decoys for the identification of native-like RNP structures</bold>. Four cases scored by the DARS-RNP potential have been selected from the analysis presented in Figure 6. The structure with the lowest energy (score) is selected as a representative of a cluster. Native structures are in dark colors (blue -RNA and red - protein), while decoys are indicated by light colors (light blue - RNA and salmon pink - protein). Bigger components are superimposed.</p></caption><graphic xlink:href="1471-2105-12-348-7"/></fig><table-wrap id="T3" position="float"><label>Table 3</label><caption><p>Results of scoring the decoys in the GRAMM-generated unbound docking set</p></caption><table frame="hsides" rules="groups"><thead><tr><th align="center">Complex PDB code</th><th align="center">Protein/<break/>RNA<break/>PDB codes</th><th align="center">RMSD threshold (decoy vs native)</th><th align="center">Number of decoys below threshold<break/>(per 10,000)</th><th align="center" colspan="4">Correlation coefficient</th></tr><tr><th/><th colspan="7"><hr/></th></tr><tr><th/><th/><th/><th/><th align="center">DARS-RNP potential (std. error)</th><th align="left">QUASI-RNP potential (std. error)</th><th align="center">Varani potential (std. error)</th><th align="center">Fernandez potential<break/>(std. error)</th></tr></thead><tbody><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="2rkj">2rkj</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="1y42">1y42</ext-link>/<break/><ext-link ext-link-type="pdb" xlink:href="1y0q">1y0q</ext-link></td><td align="center">5</td><td align="center">4</td><td align="center">-0.58 (0.576)</td><td align="center">0.17 (0.697)</td><td align="center">-0.6 (0.566)</td><td align="center">1 (0.000)</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">10</td><td align="center">15</td><td align="center">-0.25 (0.269)</td><td align="center">-0.11 (0.276)</td><td align="center">0.34 (0.261)</td><td align="center">0.31 (0.264)</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">20</td><td align="center">87</td><td align="center">0.09 (0.108)</td><td align="center">0.21 (0.106)</td><td align="center">0.06 (0.108)</td><td align="center">0.51 (0.093)</td></tr><tr><td colspan="8"><hr/></td></tr><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="2r8s">2r8s</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="2r8s">2r8s</ext-link>/<break/><ext-link ext-link-type="pdb" xlink:href="1hr2">1hr2</ext-link></td><td align="center">5</td><td align="center">3</td><td align="center">1 (0.000)</td><td align="center">1 (0.000)</td><td align="center">-1 (0.000)</td><td align="center">-1 (0.000)</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">10</td><td align="center">6</td><td align="center">0.7 (0.357)</td><td align="center">0.73 (0.342)</td><td align="center">-0.6 (0.400)</td><td align="center">-0.33 0.472)</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">20</td><td align="center">13</td><td align="center">0.72 (0.209)</td><td align="center">0.64 (0.232)</td><td align="center">-0.47 (0.266)</td><td align="center">-0.43 (0.272)</td></tr><tr><td colspan="8"><hr/></td></tr><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="1lng">1lng</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="1lng">1lng</ext-link>/<break/><ext-link ext-link-type="pdb" xlink:href="1z43">1z43</ext-link></td><td align="center">5</td><td align="center">2</td><td align="center">-</td><td align="center">-</td><td align="center">-</td><td align="center">-</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">10</td><td align="center">18</td><td align="center">0.1 (0.249)</td><td align="center">-0.15 (0.247)</td><td align="center">0.1 (0.249)</td><td align="center">0.14 (0.248)</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">20</td><td align="center">48</td><td align="center">0.73 (0.101)</td><td align="center">0.71 (0.104)</td><td align="center">0.4 (0.135)</td><td align="center">0.53 (0.125)</td></tr><tr><td colspan="8"><hr/></td></tr><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="2pxv">2pxv</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="2pxv">2pxv</ext-link>/<break/><ext-link ext-link-type="pdb" xlink:href="1cql">1cql</ext-link></td><td align="center">5</td><td align="center">14</td><td align="center">0.69 (0.209)</td><td align="center">0.02 (0.289)</td><td align="center">0.28 (0.277)</td><td align="center">-0.19 (0.283)</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">10</td><td align="center">131</td><td align="center">-0.04 (0.088)</td><td align="center">0.02 (0.088)</td><td align="center">0.14 (0.087)</td><td align="center">-0.03 (0.088)</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">20</td><td align="center">948</td><td align="center">-0.08 (0.032)</td><td align="center">-0.02 (0.033)</td><td align="center">0.13 (0.032)</td><td align="center">0.17 (0.032)</td></tr><tr><td colspan="8"><hr/></td></tr><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="1e7k">1e7k</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="2jnb">2jnb</ext-link>/<break/><ext-link ext-link-type="pdb" xlink:href="1e7k">1e7k</ext-link></td><td align="center">5</td><td align="center">7</td><td align="center">-0.08 (0.446)</td><td align="center">0.09 (0.445)</td><td align="center">0.6 (0.358)</td><td align="center">0.11 (0.444)</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">10</td><td align="center">46</td><td align="center">0.31 (0.143)</td><td align="center">0.39 (0.139)</td><td align="center">0.26 (0.146)</td><td align="center">0.2 (0.148)</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">20</td><td align="center">938</td><td align="center">0.19 (0.032)</td><td align="center">0.17 (0.032)</td><td align="center">0.14 (0.032)</td><td align="center">0.17 (0.032)</td></tr><tr><td colspan="8"><hr/></td></tr><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="1wpu">1wpu</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="1wpv">1wpv</ext-link>/<break/><ext-link ext-link-type="pdb" xlink:href="1wpu">1wpu</ext-link></td><td align="center">5</td><td align="center">125</td><td align="center">0.63 (0.070)</td><td align="center">0.67 (0.067)</td><td align="center">-0.08 (0.090)</td><td align="center">-0.3 (0.086)</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">10</td><td align="center">227</td><td align="center">0.42 (0.061)</td><td align="center">0.34 (0.063)</td><td align="center">0.02 (0.067)</td><td align="center">-0.4 (0.061)</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">20</td><td align="center">1165</td><td align="center">0.5 (0.025)</td><td align="center">0.37 (0.027)</td><td align="center">0.02 (0.029)</td><td align="center">-0.21(0.029)</td></tr><tr><td colspan="8"><hr/></td></tr><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="3bso">3bso</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="1sh0">1sh0</ext-link>/<break/><ext-link ext-link-type="pdb" xlink:href="3bso">3bso</ext-link></td><td align="center">5</td><td align="center">34</td><td align="center">0.48 (0.155)</td><td align="center">0.32 (0.167)</td><td align="center">-0.1 (0.176)</td><td align="center">-0.09 (0.176)</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">10</td><td align="center">104</td><td align="center">0.52 (0.085)</td><td align="center">0.49 (0.086)</td><td align="center">-0.03 (0.099)</td><td align="center">-0.32 (0.094)</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">20</td><td align="center">1212</td><td align="center">0.42 (0.026)</td><td align="center">0.41 (0.026)</td><td align="center">-0.21 (0.028)</td><td align="center">0.09 (0.029)</td></tr><tr><td colspan="8"><hr/></td></tr><tr><td align="center"><ext-link ext-link-type="pdb" xlink:href="2jea">2jea</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="2je6">2je6</ext-link>/<break/><ext-link ext-link-type="pdb" xlink:href="2jea">2jea</ext-link></td><td align="center">5</td><td align="center">35</td><td align="center">0.16 (0.172)</td><td align="center">0.22 (0.170)</td><td align="center">0.34 (0.164)</td><td align="center">0.44 (0.156)</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">10</td><td align="center">440</td><td align="center">0.3 (0.046)</td><td align="center">0.21 (0.047)</td><td align="center">0.29 (0.046)</td><td align="center">0.13 (0.047)</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">20</td><td align="center">3290</td><td align="center">0.41 (0.016)</td><td align="center">0.42 (0.016)</td><td align="center">0.04 (0.017)</td><td align="center">0.18 (0.017)</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">5</td><td align="center">28</td><td align="center">0.32</td><td align="center">0.26</td><td align="center">0.21</td><td align="center">-0.01</td></tr><tr><td colspan="8"><hr/></td></tr><tr><td align="center" colspan="2">Mean</td><td align="center">10</td><td align="center">123.4</td><td align="center">0.26</td><td align="center">0.23</td><td align="center">0.06</td><td align="center">-0.04</td></tr><tr><td/><td colspan="7"><hr/></td></tr><tr><td/><td/><td align="center">20</td><td align="center">962.6</td><td align="center">0.37</td><td align="center">0.34</td><td align="center">0.01</td><td align="center">0.13</td></tr></tbody></table><table-wrap-foot><p>Correlation coefficients were calculated for scatter plots from Figure 6.</p></table-wrap-foot></table-wrap></sec><sec><title>Clustering of the best-scored decoys</title><p>The application of clustering to identify groups of similar structures among the top-scored decoys improves the predictive power of all statistical potentials analyzed in this work. It helps Fernandez potential to recognize near-native structures in the Varani set for bound docking with optimization of side chains (Figure <xref ref-type="fig" rid="F4">4</xref>), our DARS-RNP and QUASI-RNP potentials in the GRAMM bound docking test set (Figure <xref ref-type="fig" rid="F5">5</xref>), and all potentials in the unbound docking test set (Figure <xref ref-type="fig" rid="F6">6</xref>). Figure <xref ref-type="fig" rid="F7">7</xref> shows examples of four complexes where the native-structure was found owing to the clustering of well-scored models identified by the DARS-RNP potential.</p></sec></sec><sec><title>Discussion</title><p>The QUASI-RNP and DARS-RNP potentials described in this work exhibit the highest discriminatory power for the bound Varani set, where there are many near-native structures without steric clashes. Likewise, our methods performed well for another set of decoys generated for the same RNA-protein complex structures with the GRAMM method. Our potentials failed to recognize native-like structures generated by GRAMM only for the 1JID structure (human SRP19 protein in complex with helix 6 of human SRP RNA). Both DARS-RNP and QUASI-RNP favored a structure that is very different from the native complex, even though they were able to recognize native-like structures for the same complex generated by Varani. The visual examination of models for the 1JID complex that were best-scored by our potentials revealed structures, in which the RNA backbone has entered a very deep and narrow groove on the protein structure far from the true RNA-binding site, leading to decoys with very big area of protein-RNA interactions, and hence with significantly more contacts than in the native structure (solvent-accessible surface area buried upon complex formation: ~2600 &#x000c5;<sup>2 </sup>vs ~1600 &#x000c5;<sup>2 </sup>for the misleading decoy and the native structure, respectively). One way to avoid such situations is to identify (or predict) the RNA-binding site on the protein surface and filter the initial decoys (e, g. using our method FILTREST3D [<xref ref-type="bibr" rid="B32">32</xref>]) to remove those with RNA away from the binding site.</p><p>It is worth to mention that the five structures in the bound docking test sets were excluded from the training set only for the QUASI-RNP and DARS-RNP potentials. The training set of the Fernandez potential contained three out of five assessed complexes, and the Varani training set contained all five complexes. Therefore, the ability of the Varani and Fernandez potentials to discriminate native-like models for these structures may be overestimated. In particular, the Varani potential has the best results for those decoys from the GRAMM-generated unbound set that are very close to native structure (RMSD &#x0003c; 5 &#x000c5;). There, the Varani potential easily recognizes the structures close to those in its training set. Interestingly, for decoys from the same set, with RMSD up to 10 or 20 &#x000c5;, our potentials still exhibit better results than the Varani potential, suggesting that they do have a power to discriminate between these medium-quality decoys and those that are totally native-unlike.</p><p>The results of tests for the unbound docking set are more objective due to a complete separation of training and testing data, and because they simulated the predictive power of the potentials in a real docking experiment, where the bound conformations of components are unknown. Among the four methods tested, QUASI-RNP and DARS-RNP potentials have the biggest average correlation coefficients between the scores and the RMSD of the model from the native structure. However, it must be emphasized that even these "winners" of our benchmark were able to identify native-like structures only for four out of eight tested cases (Figure <xref ref-type="fig" rid="F6">6</xref> and <xref ref-type="fig" rid="F7">7</xref>). As expected, our potentials recognized near native structures only for these complexes, whose components exhibit relatively small structural changes (RMSD &#x0003c; 3 &#x000c5;) during complex formation (Table <xref ref-type="table" rid="T4">4</xref>). Still, in our hands, the Fernandez and Varani potentials recognized near-native structures for one complex only with the default options of clustering (100 best-scored decoys, RMSD threshold of 5 &#x000c5;), and three and two complexes, respectively, after increasing the number of clustered best-scored decoys to 200 and the RMSD threshold to 10 &#x000c5; (Figure <xref ref-type="fig" rid="F6">6</xref> and Table <xref ref-type="table" rid="T3">3</xref>). Such "relaxed" clusters are of course more heterogeneous. For two complexes (2JEA and 1LNG) all four potentials considered in this work have recognized native-like structures. For both of these complexes only one component of the complex was solved in isolation from the other (hence it was actually half-bound/half-unbound docking) and that 'unbound' component underwent only a very minor conformational change with respect to the bound form (RMSD &#x0003c; 3 &#x000c5;) (Table <xref ref-type="table" rid="T4">4</xref>). Hence, these two targets must be considered as very easy.</p><table-wrap id="T4" position="float"><label>Table 4</label><caption><p>Complexes considered in the unbound docking experiment, for which GRAMM produced near-native structures, with their PDB and chain identifiers</p></caption><table frame="hsides" rules="groups"><thead><tr><th align="center">Recognition by DARS-RNP and QUASI-RNP</th><th align="center">Molecule name</th><th align="center">Complex PDB code</th><th align="center">Receptor PDB code (and its RMSD vs the bound structure)</th><th align="center">Ligand PDB code (and its RMSD vs the bound structure)</th></tr></thead><tbody><tr><td/><td align="center">Norwalk Virus polymerase with CTP/RNA primer</td><td align="center"><ext-link ext-link-type="pdb" xlink:href="3bso_a:p">3bso_a:p</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="1sh0_b (1.3)">1sh0_b (1.3)</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="3bso_p (0.0)">3bso_p (0.0)</ext-link></td></tr><tr><td/><td colspan="4"><hr/></td></tr><tr><td align="center">Successful</td><td align="center">HutP/Hut mRNA</td><td align="center"><ext-link ext-link-type="pdb" xlink:href="1wpu_a:c">1wpu_a:c</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="1wpv_a (0.2)">1wpv_a (0.2)</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="1wpu_c (0.0)">1wpu_c (0.0)</ext-link></td></tr><tr><td/><td colspan="4"><hr/></td></tr><tr><td/><td align="center">9-subunit archaeal exosome/RNA</td><td align="center"><ext-link ext-link-type="pdb" xlink:href="2jea_a, b:c">2jea_a, b:c</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="2je6_a, b (0.4)">2je6_a, b (0.4)</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="2jea_c (0.0)">2jea_c (0.0)</ext-link></td></tr><tr><td/><td colspan="4"><hr/></td></tr><tr><td/><td align="center">SRP 19/7S.S SRP RNA</td><td align="center"><ext-link ext-link-type="pdb" xlink:href="1lng_a:b">1lng_a:b</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="1lng_a (0.0)">1lng_a (0.0)</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="1z43_a (2.1)">1z43_a (2.1)</ext-link></td></tr><tr><td colspan="5"><hr/></td></tr><tr><td/><td align="center">Tyrosyl - tRNA synthetase splicing factor/group I intron RNA</td><td align="center"><ext-link ext-link-type="pdb" xlink:href="2rkj_b:c">2rkj_b:c</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="1y0q_a (3.0)">1y0q_a (3.0)</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="1y42_x (0.9)">1y42_x (0.9)</ext-link></td></tr><tr><td/><td colspan="4"><hr/></td></tr><tr><td align="center">Unsuccessful</td><td align="center">SRP C-termina domain/4.5 S RNA</td><td align="center"><ext-link ext-link-type="pdb" xlink:href="2pxv_a:b">2pxv_a:b</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="1cql_a (8.1)">1cql_a (8.1)</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="2pxv_a (0.0)">2pxv_a (0.0)</ext-link></td></tr><tr><td/><td colspan="4"><hr/></td></tr><tr><td/><td align="center">spliceosomal 15.5 K protein/U4 snRNA fragment</td><td align="center"><ext-link ext-link-type="pdb" xlink:href="1e7k_a:c">1e7k_a:c</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="2jnb_a (3.2)">2jnb_a (3.2)</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="1e7k_c (0.0)">1e7k_c (0.0)</ext-link></td></tr><tr><td/><td colspan="4"><hr/></td></tr><tr><td/><td align="center">Synthetic Fab/P4-P6 ribozyme domain</td><td align="center"><ext-link ext-link-type="pdb" xlink:href="2r8s_l, h:r">2r8s_l, h:r</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="1hr2_a (4.3)">1hr2_a (4.3)</ext-link></td><td align="center"><ext-link ext-link-type="pdb" xlink:href="2r8s_l, h (0.0)">2r8s_l, h (0.0)</ext-link></td></tr></tbody></table></table-wrap><p>Among the four knowledge-based potentials tested here for their ability to identify native-like protein-RNA docking models, the high-resolution Varani potential exhibited the best ability to recognize models closest to the native structure (RMSD &#x0003c; 5 &#x000c5;). This potential appears as the method of choice for high-accuracy docking methods that are able to generate structures very close to the native ones. It must be emphasized, however, that there are no computational tools, with which to reliably predict conformational changes upon binding. Therefore high-resolution docking is only applicable in situations, where the receptor and ligand structures exhibit very little conformational change between the bound and unbound forms. Unfortunately, whether the conformational change occurs or not cannot be reliably predicted. In most cases of protein-RNA binding, moderate conformational changes of protein and/or RNA molecules occur upon complex formation. There, low resolution methods that apply a coarse-grained energy model to a coarse-grained representation (i.e. without looking at the atomic details that change upon binding) have a chance to be practically useful.</p><p>Among the four potentials tested in this work, the Fernandez potential has the weakest discriminatory power for identification of near-native structures, but we believe this potential may perform much better when combined with the FTDock scoring function, as originally intended by the Fernandez group. However, the combination of Fernandez and FTDock potentials is possible only for FTDock decoys, because the FTDock potential is not available as a standalone program, hence we could not apply it to the decoys generated in our study. FTDock is also unable to deal with modified residues in the RNA, which precludes it from applicability to many RNA-protein complexes, where RNA modifications play a critical role (e.g. interactions of tRNA with aminoacyl-tRNA synthetases).</p><p>The main difference between all potentials considered in this article is the (sub)set of atoms taken into consideration. The Varani potential considers interactions between all atoms, with chemically similar atoms (based on the CHARMM atom definition) treated in the same way. It contains only a distance-dependent multiple bin term. The Fernandez potential calculates interactions between entire residues represented as single interaction centers, using only one bin (i.e. the interaction is either present or absent). Our potentials use multiple bins for distance as well as orientation, hence they take into account more information about the possible arrangements of amino acid and nucleotide residues, even though they use less atoms than the Varani method.</p><p>In our study we have used both bound and unbound conformations for docking (with bound structures either completely unmodified, as in the GRAMM experiment, or with side-chains repacked, as in the Varani experiment). The shapes of score-RMSD dependence plots show differences associated not only with the methods, but also with the type of docking. As expected, all potentials exhibit best results for bound docking experiment, where there are many near-native structures without steric clashes. This observation underlines the influence of decoy generation method on the ability to successfully identify native-like decoys in the generated dataset.</p><p>Our study allows for direct comparison of the decoy-based and quasi-chemical approaches for calculating statistical potentials. The small, but significant average advantage of the DARS-RNP potential over the QUASI-RNP potential can be explained by the more realistic treatment of "random" protein-RNA interactions. In the DARS-based approach, the statistics of amino acid-nucleotide contacts are inferred from geometrically plausible, but biologically irrelevant decoys (pseudo-complexes), while the quasi-chemical approach predicts the occurrence of such contacts based on the frequency of individual residues. We expect this advantage of the DARS approach to hold also for other types of docking. The calculation of a DARS-based potential requires, however, the calculation of a large number of decoys for each complex in the training set, hence it requires considerably bigger computational effort, which may be prohibitive in case of large training sets.</p><p>By definition, none of the rigid-body docking methods analyzed here is capable of predicting the structures of complexes that involve large conformational changes. We also found that the presence of extensive steric clashes in decoys deteriorates the discriminatory power of all potentials tested in our benchmark. Thus, we propose that the next step in the development of methods for modeling of protein-RNA complexes should be taken towards algorithms that enable simultaneous docking and (re)folding of protein and RNA components. Recently, a number of methods for modeling of RNA 3D structures have been reported that utilize very similar methodology to that used for protein modeling [<xref ref-type="bibr" rid="B6">6</xref>,<xref ref-type="bibr" rid="B33">33</xref>]. This suggests that the combination of comparative modeling and "de novo" folding should be possible not only for proteins and RNA separately, but also as components of the same molecular system.</p></sec><sec><title>Conclusions</title><p>Among the four potentials tested in this work the QUASI-RNP and DARS-RNP potentials exhibit the highest discriminatory power for both bound and unbound docking experiments. The small average advantage of the DARS-RNP potential over the QUASI-RNP potential can be caused by the more realistic treatment of "random" protein-RNA interactions. None of the rigid-body docking methods analyzed here is capable of predicting the structures of complexes that involve large conformational changes. Our potentials recognized near native structures only for these complexes, whose components exhibit relatively small structural changes (RMSD &#x0003c; 3 &#x000c5;) during formation of the protein-RNA complex.</p><sec><title>Availability and requirements</title><p>Project name: Statistical potentials for Protein-RNA docking; Project home page: <ext-link ext-link-type="uri" xlink:href="http://iimcb.genesilico.pl/RNP/">http://iimcb.genesilico.pl/RNP/</ext-link>; Operating system: Linux; Programming language: Python; Other requirements: Biopython and NumPy libraries; Licence: GNU GPL; No restrictions to use by non-academics.</p></sec></sec><sec><title>List of abbreviations</title><p>RMSD: Root Mean Square Deviation; RNP: RNA-protein complexes; QUASI-RNP: QUASI-chemical potential for RNA-Protein complexes; DARS-RNP: Decoys As Reference State potential for RNA-Protein complexes.</p></sec><sec><title>Authors' contributions</title><p>IT developed statistical potentials, performed testing procedures, drafted the manuscript and prepared the figures. JMB conceived of the project and edited the manuscript. Both authors analyzed and interpreted the data. All authors read and approved the final manuscript.</p></sec><sec sec-type="supplementary-material"><title>Supplementary Material</title><supplementary-material content-type="local-data" id="S1"><caption><title>Additional file 1</title><p><bold>Additional_figures_and_table.pdf</bold>. Files and tables with additional data illustrating the details of implementation, ordered according to their appearance in the text.</p></caption><media xlink:href="1471-2105-12-348-S1.PDF" mimetype="application" mime-subtype="pdf"><caption><p>Click here for file</p></caption></media></supplementary-material><supplementary-material content-type="local-data" id="S2"><caption><title>Additional file 2</title><p><bold>Energy_DARS.pdf</bold>. Energy for each distance, angle, and site bin, for each pair wise interaction in the DARS-RNP potential.</p></caption><media xlink:href="1471-2105-12-348-S2.PDF" mimetype="application" mime-subtype="pdf"><caption><p>Click here for file</p></caption></media></supplementary-material><supplementary-material content-type="local-data" id="S3"><caption><title>Additional file 3</title><p><bold>Energy_QUASI.pdf</bold>. Energy for each distance, angle, and site bin, for each pair wise interaction in the QUASI-RNP potential.</p></caption><media xlink:href="1471-2105-12-348-S3.PDF" mimetype="application" mime-subtype="pdf"><caption><p>Click here for file</p></caption></media></supplementary-material><supplementary-material content-type="local-data" id="S4"><caption><title>Additional file 4</title><p><bold>Observed.pdf</bold>. Observed number of contacts in each distance, angle, and site bin, for each pair wise interaction.</p></caption><media xlink:href="1471-2105-12-348-S4.PDF" mimetype="application" mime-subtype="pdf"><caption><p>Click here for file</p></caption></media></supplementary-material><supplementary-material content-type="local-data" id="S5"><caption><title>Additional file 5</title><p><bold>Expected_DARS.pdf</bold>. Expected number of contacts in each distance, angle, and site bin, for each pair wise interaction in the DARS-RNP potential.</p></caption><media xlink:href="1471-2105-12-348-S5.PDF" mimetype="application" mime-subtype="pdf"><caption><p>Click here for file</p></caption></media></supplementary-material><supplementary-material content-type="local-data" id="S6"><caption><title>Additional file 6</title><p><bold>Expected_QUASI.pdf</bold>. Expected number of contacts in each distance, angle, and site bin, for each pair wise interaction in the QUASI-RNP potential.</p></caption><media xlink:href="1471-2105-12-348-S6.PDF" mimetype="application" mime-subtype="pdf"><caption><p>Click here for file</p></caption></media></supplementary-material></sec></body><back><sec><title>Acknowledgements and Funding</title><p>We thank Michal Boniecki and Marcin Pawlowski for inspiration and discussions and Michal Pietal for sharing BrutePDBParser. We also wish to thank Kristian Rother, Grzegorz Chojnowski, Anna Philips, and Lukasz Kozlowski for critical reading of the manuscript. We thank Gabriele Varani and Juan Fernandez-Recio for providing us with their software and data for comparison with our methods. This project was funded by the Foundation for Polish Science (FNP, grant TEAM/2009-4/2). I.T. was supported by the 6th Framework Programme of the European Commission (EC FP6, Network of Excellence EURANSET, contract number LSHG-CT-2005-518238). J.M.B. was supported by the European Research Council (ERC, StG grant RNA+P = 123D) and by the "Ideas for Poland" fellowship from the Foundation of Polish Science (FNP).</p></sec><ref-list><ref id="B1"><mixed-citation publication-type="journal"><name><surname>Chen</surname><given-names>Y</given-names></name><name><surname>Varani</surname><given-names>G</given-names></name><article-title>Protein families and RNA recognition</article-title><source>Febs J</source><year>2005</year><volume>272</volume><issue>9</issue><fpage>2088</fpage><lpage>2097</lpage><pub-id pub-id-type="doi">10.1111/j.1742-4658.2005.04650.x</pub-id><pub-id pub-id-type="pmid">15853794</pub-id></mixed-citation></ref><ref id="B2"><mixed-citation publication-type="journal"><name><surname>Lukong</surname><given-names>KE</given-names></name><name><surname>Chang</surname><given-names>KW</given-names></name><name><surname>Khandjian</surname><given-names>EW</given-names></name><name><surname>Richard</surname><given-names>S</given-names></name><article-title>RNA-binding proteins in human genetic disease</article-title><source>Trends Genet</source><year>2008</year><volume>24</volume><issue>8</issue><fpage>416</fpage><lpage>425</lpage><pub-id pub-id-type="doi">10.1016/j.tig.2008.05.004</pub-id><pub-id pub-id-type="pmid">18597886</pub-id></mixed-citation></ref><ref id="B3"><mixed-citation publication-type="journal"><name><surname>Ke</surname><given-names>A</given-names></name><name><surname>Doudna</surname><given-names>JA</given-names></name><article-title>Crystallization of RNA and RNA-protein complexes</article-title><source>Methods</source><year>2004</year><volume>34</volume><issue>3</issue><fpage>408</fpage><lpage>414</lpage><pub-id pub-id-type="doi">10.1016/j.ymeth.2004.03.027</pub-id><pub-id pub-id-type="pmid">15325657</pub-id></mixed-citation></ref><ref id="B4"><mixed-citation publication-type="journal"><name><surname>Scott</surname><given-names>LG</given-names></name><name><surname>Hennig</surname><given-names>M</given-names></name><article-title>RNA structure determination by NMR</article-title><source>Methods Mol Biol</source><year>2008</year><volume>452</volume><fpage>29</fpage><lpage>61</lpage><pub-id pub-id-type="doi">10.1007/978-1-60327-159-2_2</pub-id><pub-id pub-id-type="pmid">18563368</pub-id></mixed-citation></ref><ref id="B5"><mixed-citation publication-type="journal"><name><surname>Laing</surname><given-names>C</given-names></name><name><surname>Schlick</surname><given-names>T</given-names></name><article-title>Computational approaches to 3D modeling of RNA</article-title><source>J Phys Condens Matter</source><year>2010</year><volume>22</volume><issue>28</issue><fpage>283101</fpage><pub-id pub-id-type="doi">10.1088/0953-8984/22/28/283101</pub-id><pub-id pub-id-type="pmid">21399271</pub-id></mixed-citation></ref><ref id="B6"><mixed-citation publication-type="journal"><name><surname>Rother</surname><given-names>K</given-names></name><name><surname>Rother</surname><given-names>M</given-names></name><name><surname>Boniecki</surname><given-names>M</given-names></name><name><surname>Puton</surname><given-names>T</given-names></name><name><surname>Bujnicki</surname><given-names>JM</given-names></name><article-title>RNA and protein 3D structure modeling: similarities and differences</article-title><source>J Mol Model</source><year>2011</year><volume>17</volume><issue>9</issue><fpage>2325</fpage><lpage>2336</lpage><pub-id pub-id-type="doi">10.1007/s00894-010-0951-x</pub-id><pub-id pub-id-type="pmid">21258831</pub-id></mixed-citation></ref><ref id="B7"><mixed-citation publication-type="journal"><name><surname>Moreira</surname><given-names>IS</given-names></name><name><surname>Fernandes</surname><given-names>PA</given-names></name><name><surname>Ramos</surname><given-names>MJ</given-names></name><article-title>Protein-protein docking dealing with the unknown</article-title><source>J Comp Chem</source><year>2010</year><volume>31</volume><issue>2</issue><fpage>317</fpage><lpage>342</lpage></mixed-citation></ref><ref id="B8"><mixed-citation publication-type="journal"><name><surname>Zacharias</surname><given-names>M</given-names></name><article-title>Accounting for conformational changes during protein-protein docking</article-title><source>Curr Opin Struct Biol</source><year>2010</year><volume>20</volume><issue>2</issue><fpage>180</fpage><lpage>186</lpage><pub-id pub-id-type="doi">10.1016/j.sbi.2010.02.001</pub-id><pub-id pub-id-type="pmid">20194014</pub-id></mixed-citation></ref><ref id="B9"><mixed-citation publication-type="journal"><name><surname>Dunin-Horkawicz</surname><given-names>S</given-names></name><name><surname>Czerwoniec</surname><given-names>A</given-names></name><name><surname>Gajda</surname><given-names>MJ</given-names></name><name><surname>Feder</surname><given-names>M</given-names></name><name><surname>Grosjean</surname><given-names>H</given-names></name><name><surname>Bujnicki</surname><given-names>JM</given-names></name><article-title>MODOMICS: a database of RNA modification pathways</article-title><source>Nucleic Acids Res</source><year>2006</year><volume>34</volume><issue>Database</issue><fpage>D145</fpage><lpage>149</lpage><pub-id pub-id-type="pmid">16381833</pub-id></mixed-citation></ref><ref id="B10"><mixed-citation publication-type="other"><name><surname>Grosjean</surname><given-names>H</given-names></name><article-title>Fine-Tuning of RNA Functions by Modification and Editing</article-title><year>2009</year></mixed-citation></ref><ref id="B11"><mixed-citation publication-type="journal"><name><surname>Mucha</surname><given-names>P</given-names></name><name><surname>Szyk</surname><given-names>A</given-names></name><name><surname>Rekowski</surname><given-names>P</given-names></name><name><surname>Weiss</surname><given-names>PA</given-names></name><name><surname>Agris</surname><given-names>PF</given-names></name><article-title>Anticodon domain methylated nucleosides of yeast tRNA(Phe) are significant recognition determinants in the binding of a phage display selected peptide</article-title><source>Biochemistry</source><year>2001</year><volume>40</volume><issue>47</issue><fpage>14191</fpage><lpage>14199</lpage><pub-id pub-id-type="doi">10.1021/bi010978o</pub-id><pub-id pub-id-type="pmid">11714272</pub-id></mixed-citation></ref><ref id="B12"><mixed-citation publication-type="journal"><name><surname>Soma</surname><given-names>A</given-names></name><name><surname>Ikeuchi</surname><given-names>Y</given-names></name><name><surname>Kanemasa</surname><given-names>S</given-names></name><name><surname>Kobayashi</surname><given-names>K</given-names></name><name><surname>Ogasawara</surname><given-names>N</given-names></name><name><surname>Ote</surname><given-names>T</given-names></name><name><surname>Kato</surname><given-names>J</given-names></name><name><surname>Watanabe</surname><given-names>K</given-names></name><name><surname>Sekine</surname><given-names>Y</given-names></name><name><surname>Suzuki</surname><given-names>T</given-names></name><article-title>An RNA-modifying enzyme that governs both the codon and amino acid specificities of isoleucine tRNA</article-title><source>Mol Cell</source><year>2003</year><volume>12</volume><issue>3</issue><fpage>689</fpage><lpage>698</lpage><pub-id pub-id-type="doi">10.1016/S1097-2765(03)00346-0</pub-id><pub-id pub-id-type="pmid">14527414</pub-id></mixed-citation></ref><ref id="B13"><mixed-citation publication-type="journal"><name><surname>Dominguez</surname><given-names>C</given-names></name><name><surname>Boelens</surname><given-names>R</given-names></name><name><surname>Bonvin</surname><given-names>AM</given-names></name><article-title>HADDOCK: a protein-protein docking approach based on biochemical or biophysical information</article-title><source>J Am Chem Soc</source><year>2003</year><volume>125</volume><issue>7</issue><fpage>1731</fpage><lpage>1737</lpage><pub-id pub-id-type="doi">10.1021/ja026939x</pub-id><pub-id pub-id-type="pmid">12580598</pub-id></mixed-citation></ref><ref id="B14"><mixed-citation publication-type="journal"><name><surname>Vakser</surname><given-names>IA</given-names></name><name><surname>Aflalo</surname><given-names>C</given-names></name><article-title>Hydrophobic docking: a proposed enhancement to molecular recognition techniques</article-title><source>Proteins</source><year>1994</year><volume>20</volume><issue>4</issue><fpage>320</fpage><lpage>329</lpage><pub-id pub-id-type="doi">10.1002/prot.340200405</pub-id><pub-id pub-id-type="pmid">7731951</pub-id></mixed-citation></ref><ref id="B15"><mixed-citation publication-type="journal"><name><surname>Ritchie</surname><given-names>DW</given-names></name><name><surname>Kemp</surname><given-names>GJ</given-names></name><article-title>Protein docking using spherical polar Fourier correlations</article-title><source>Proteins</source><year>2000</year><volume>39</volume><issue>2</issue><fpage>178</fpage><lpage>194</lpage><pub-id pub-id-type="doi">10.1002/(SICI)1097-0134(20000501)39:2&#x0003c;178::AID-PROT8&#x0003e;3.0.CO;2-6</pub-id><pub-id pub-id-type="pmid">10737939</pub-id></mixed-citation></ref><ref id="B16"><mixed-citation publication-type="journal"><name><surname>Zheng</surname><given-names>S</given-names></name><name><surname>Robertson</surname><given-names>TA</given-names></name><name><surname>Varani</surname><given-names>G</given-names></name><article-title>A knowledge-based potential function predicts the specificity and relative binding energy of RNA-binding proteins</article-title><source>Febs J</source><year>2007</year><volume>274</volume><issue>24</issue><fpage>6378</fpage><lpage>6391</lpage><pub-id pub-id-type="pmid">18005254</pub-id></mixed-citation></ref><ref id="B17"><mixed-citation publication-type="other"><name><surname>Perez-Cano</surname><given-names>L</given-names></name><name><surname>Solernou</surname><given-names>A</given-names></name><name><surname>Pons</surname><given-names>C</given-names></name><name><surname>Fernandez-Recio</surname><given-names>J</given-names></name><article-title>Structural prediction of protein-RNA interaction by computational docking with propensity-based statistical potentials</article-title><source>Pac Symp Biocomput</source><year>2010</year><fpage>293</fpage><lpage>301</lpage></mixed-citation></ref><ref id="B18"><mixed-citation publication-type="journal"><name><surname>Gabb</surname><given-names>HA</given-names></name><name><surname>Jackson</surname><given-names>RM</given-names></name><name><surname>Sternberg</surname><given-names>MJ</given-names></name><article-title>Modelling protein docking using shape complementarity, electrostatics and biochemical information</article-title><source>J Mol Biol</source><year>1997</year><volume>272</volume><issue>1</issue><fpage>106</fpage><lpage>120</lpage><pub-id pub-id-type="doi">10.1006/jmbi.1997.1203</pub-id><pub-id pub-id-type="pmid">9299341</pub-id></mixed-citation></ref><ref id="B19"><mixed-citation publication-type="journal"><name><surname>Chuang</surname><given-names>GY</given-names></name><name><surname>Kozakov</surname><given-names>D</given-names></name><name><surname>Brenke</surname><given-names>R</given-names></name><name><surname>Comeau</surname><given-names>SR</given-names></name><name><surname>Vajda</surname><given-names>S</given-names></name><article-title>DARS (Decoys As the Reference State) potentials for protein-protein docking</article-title><source>Biophys J</source><year>2008</year><volume>95</volume><issue>9</issue><fpage>4217</fpage><lpage>4227</lpage><pub-id pub-id-type="doi">10.1529/biophysj.108.135814</pub-id><pub-id pub-id-type="pmid">18676649</pub-id></mixed-citation></ref><ref id="B20"><mixed-citation publication-type="journal"><name><surname>Katchalski-Katzir</surname><given-names>E</given-names></name><name><surname>Shariv</surname><given-names>I</given-names></name><name><surname>Eisenstein</surname><given-names>M</given-names></name><name><surname>Friesem</surname><given-names>AA</given-names></name><name><surname>Aflalo</surname><given-names>C</given-names></name><name><surname>Vakser</surname><given-names>IA</given-names></name><article-title>Molecular surface recognition: determination of geometric fit between proteins and their ligands by correlation techniques</article-title><source>Proc Natl Acad Sci USA</source><year>1992</year><volume>89</volume><issue>6</issue><fpage>2195</fpage><lpage>2199</lpage><pub-id pub-id-type="doi">10.1073/pnas.89.6.2195</pub-id><pub-id pub-id-type="pmid">1549581</pub-id></mixed-citation></ref><ref id="B21"><mixed-citation publication-type="journal"><name><surname>Donald</surname><given-names>JE</given-names></name><name><surname>Chen</surname><given-names>WW</given-names></name><name><surname>Shakhnovich</surname><given-names>EI</given-names></name><article-title>Energetics of protein-DNA interactions</article-title><source>Nucleic Acids Res</source><year>2007</year><volume>35</volume><issue>4</issue><fpage>1039</fpage><lpage>1047</lpage><pub-id pub-id-type="doi">10.1093/nar/gkl1103</pub-id><pub-id pub-id-type="pmid">17259221</pub-id></mixed-citation></ref><ref id="B22"><mixed-citation publication-type="journal"><name><surname>Leontis</surname><given-names>NB</given-names></name><name><surname>Westhof</surname><given-names>E</given-names></name><article-title>Geometric nomenclature and classification of RNA base pairs</article-title><source>Rna</source><year>2001</year><volume>7</volume><issue>4</issue><fpage>499</fpage><lpage>512</lpage><pub-id pub-id-type="doi">10.1017/S1355838201002515</pub-id><pub-id pub-id-type="pmid">11345429</pub-id></mixed-citation></ref><ref id="B23"><mixed-citation publication-type="journal"><name><surname>Boniecki</surname><given-names>M</given-names></name><name><surname>Rotkiewicz</surname><given-names>P</given-names></name><name><surname>Skolnick</surname><given-names>J</given-names></name><name><surname>Kolinski</surname><given-names>A</given-names></name><article-title>Protein fragment reconstruction using various modeling techniques</article-title><source>J Comput Aided Mol Des</source><year>2003</year><volume>17</volume><issue>11</issue><fpage>725</fpage><lpage>738</lpage><pub-id pub-id-type="pmid">15072433</pub-id></mixed-citation></ref><ref id="B24"><mixed-citation publication-type="journal"><name><surname>Malolepsza</surname><given-names>E</given-names></name><name><surname>Boniecki</surname><given-names>M</given-names></name><name><surname>Kolinski</surname><given-names>A</given-names></name><name><surname>Piela</surname><given-names>L</given-names></name><article-title>Theoretical model of prion propagation: a misfolded protein induces misfolding</article-title><source>Proc Natl Acad Sci USA</source><year>2005</year><volume>102</volume><issue>22</issue><fpage>7835</fpage><lpage>7840</lpage><pub-id pub-id-type="doi">10.1073/pnas.0409389102</pub-id><pub-id pub-id-type="pmid">15911770</pub-id></mixed-citation></ref><ref id="B25"><mixed-citation publication-type="journal"><name><surname>Berman</surname><given-names>HM</given-names></name><name><surname>Westbrook</surname><given-names>J</given-names></name><name><surname>Feng</surname><given-names>Z</given-names></name><name><surname>Gilliland</surname><given-names>G</given-names></name><name><surname>Bhat</surname><given-names>TN</given-names></name><name><surname>Weissig</surname><given-names>H</given-names></name><name><surname>Shindyalov</surname><given-names>IN</given-names></name><name><surname>Bourne</surname><given-names>PE</given-names></name><article-title>The Protein Data Bank</article-title><source>Nucleic Acids Res</source><year>2000</year><volume>28</volume><issue>1</issue><fpage>235</fpage><lpage>242</lpage><pub-id pub-id-type="doi">10.1093/nar/28.1.235</pub-id><pub-id pub-id-type="pmid">10592235</pub-id></mixed-citation></ref><ref id="B26"><mixed-citation publication-type="journal"><name><surname>Chen</surname><given-names>Y</given-names></name><name><surname>Kortemme</surname><given-names>T</given-names></name><name><surname>Robertson</surname><given-names>T</given-names></name><name><surname>Baker</surname><given-names>D</given-names></name><name><surname>Varani</surname><given-names>G</given-names></name><article-title>A new hydrogen-bonding potential for the design of protein-RNA interactions predicts specific contacts and discriminates decoys</article-title><source>Nucleic Acids Res</source><year>2004</year><volume>32</volume><issue>17</issue><fpage>5147</fpage><lpage>5162</lpage><pub-id pub-id-type="doi">10.1093/nar/gkh785</pub-id><pub-id pub-id-type="pmid">15459285</pub-id></mixed-citation></ref><ref id="B27"><mixed-citation publication-type="journal"><name><surname>Gray</surname><given-names>JJ</given-names></name><name><surname>Moughon</surname><given-names>SE</given-names></name><name><surname>Kortemme</surname><given-names>T</given-names></name><name><surname>Schueler-Furman</surname><given-names>O</given-names></name><name><surname>Misura</surname><given-names>KM</given-names></name><name><surname>Morozov</surname><given-names>AV</given-names></name><name><surname>Baker</surname><given-names>D</given-names></name><article-title>Protein-protein docking predictions for the CAPRI experiment</article-title><source>Proteins</source><year>2003</year><volume>52</volume><issue>1</issue><fpage>118</fpage><lpage>122</lpage><pub-id pub-id-type="doi">10.1002/prot.10384</pub-id><pub-id pub-id-type="pmid">12784377</pub-id></mixed-citation></ref><ref id="B28"><mixed-citation publication-type="journal"><name><surname>Gray</surname><given-names>JJ</given-names></name><name><surname>Moughon</surname><given-names>S</given-names></name><name><surname>Wang</surname><given-names>C</given-names></name><name><surname>Schueler-Furman</surname><given-names>O</given-names></name><name><surname>Kuhlman</surname><given-names>B</given-names></name><name><surname>Rohl</surname><given-names>CA</given-names></name><name><surname>Baker</surname><given-names>D</given-names></name><article-title>Protein-protein docking with simultaneous optimization of rigid-body displacement and side-chain conformations</article-title><source>J Mol Biol</source><year>2003</year><volume>331</volume><issue>1</issue><fpage>281</fpage><lpage>299</lpage><pub-id pub-id-type="doi">10.1016/S0022-2836(03)00670-3</pub-id><pub-id pub-id-type="pmid">12875852</pub-id></mixed-citation></ref><ref id="B29"><mixed-citation publication-type="journal"><name><surname>Lundstrom</surname><given-names>J</given-names></name><name><surname>Rychlewski</surname><given-names>L</given-names></name><name><surname>Bujnicki</surname><given-names>J</given-names></name><name><surname>Elofsson</surname><given-names>A</given-names></name><article-title>Pcons: a neural-network-based consensus predictor that improves fold recognition</article-title><source>Protein Sci</source><year>2001</year><volume>10</volume><issue>11</issue><fpage>2354</fpage><lpage>2362</lpage><pub-id pub-id-type="pmid">11604541</pub-id></mixed-citation></ref><ref id="B30"><mixed-citation publication-type="journal"><name><surname>Cozzetto</surname><given-names>D</given-names></name><name><surname>Kryshtafovych</surname><given-names>A</given-names></name><name><surname>Tramontano</surname><given-names>A</given-names></name><article-title>Evaluation of CASP8 model quality predictions</article-title><source>Proteins</source><year>2009</year><volume>77</volume><issue>Suppl 9</issue><fpage>157</fpage><lpage>166</lpage><pub-id pub-id-type="pmid">19714774</pub-id></mixed-citation></ref><ref id="B31"><mixed-citation publication-type="journal"><name><surname>Shortle</surname><given-names>D</given-names></name><name><surname>Simons</surname><given-names>KT</given-names></name><name><surname>Baker</surname><given-names>D</given-names></name><article-title>Clustering of low-energy conformations near the native structures of small proteins</article-title><source>Proc Natl Acad Sci USA</source><year>1998</year><volume>95</volume><issue>19</issue><fpage>11158</fpage><lpage>11162</lpage><pub-id pub-id-type="doi">10.1073/pnas.95.19.11158</pub-id><pub-id pub-id-type="pmid">9736706</pub-id></mixed-citation></ref><ref id="B32"><mixed-citation publication-type="journal"><name><surname>Gajda</surname><given-names>MJ</given-names></name><name><surname>Tuszynska</surname><given-names>I</given-names></name><name><surname>Kaczor</surname><given-names>M</given-names></name><name><surname>Bakulina</surname><given-names>AY</given-names></name><name><surname>Bujnicki</surname><given-names>JM</given-names></name><article-title>FILTREST3D: discrimination of structural models using restraints from experimental data</article-title><source>Bioinformatics</source><year>2010</year><volume>26</volume><issue>23</issue><fpage>2986</fpage><lpage>2987</lpage><pub-id pub-id-type="doi">10.1093/bioinformatics/btq582</pub-id><pub-id pub-id-type="pmid">20956242</pub-id></mixed-citation></ref><ref id="B33"><mixed-citation publication-type="journal"><name><surname>Rother</surname><given-names>M</given-names></name><name><surname>Rother</surname><given-names>K</given-names></name><name><surname>Puton</surname><given-names>T</given-names></name><name><surname>Bujnicki</surname><given-names>JM</given-names></name><article-title>ModeRNA: a tool for comparative modeling of RNA 3D structure</article-title><source>Nucleic Acids Res</source><year>2011</year><volume>39</volume><issue>10</issue><fpage>4007</fpage><lpage>4022</lpage><pub-id pub-id-type="doi">10.1093/nar/gkq1320</pub-id><pub-id pub-id-type="pmid">21300639</pub-id></mixed-citation></ref></ref-list></back></article>

"""
from bs4 import BeautifulSoup
soup=BeautifulSoup(tester)


def ack1():

        from bs4 import BeautifulSoup
        import os
        import csv


        outfile = open("AckList.csv","a") #, encoding="utf-8") # <-- commented out because code was originally written for python 3.2 but then discovered that it broke 2.7


        w=csv.writer(outfile)
        w.writerow(["PMID ", "PERSONS"])

        infile=open("FinalAcks-withtaggedNEs.csv","r")
        datareader=csv.reader(infile)
#        data=[]
        for row in datareader:
                pmid=row[1]
                persons=" "
                ack=BeautifulSoup(row[3])
                if ack.findAll("person"):
                        
                        people=ack.findAll("person")
                        for i in people:
                                dude=i.get_text().encode("latin-","ignore")
                                persons=persons + dude + " , "
                print(pmid, str(persons) )

                w.writerow([pmid, str(persons)])

